auth/service_test.go

```diff
@@ -0,0 +1,192 @@
+package auth_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	testsupport "github.com/fabric8-services/fabric8-tenant/test"
+	"github.com/fabric8-services/fabric8-tenant/test/doubles"
+	"github.com/fabric8-services/fabric8-tenant/test/recorder"
+	goajwt "github.com/goadesign/goa/middleware/security/jwt"
+)
+
+func TestResolveUserToken(t *testing.T) {
+	// given
+	authService, cleanup := testdoubles.NewAuthService(t, "../test/data/token/auth_resolve_target_token", "http://authservice", recorder.WithJWTMatcher)
+	defer cleanup()
+	tok, err := testsupport.NewToken(
+		map[string]interface{}{
+			"sub": "user_foo",
+		},
+		"../test/private_key.pem",
+	)
+	require.NoError(t, err)
+
+	t.Run("ok", func(t *testing.T) {
+		// when
+		username, accessToken, err := authService.ResolveUserToken(context.Background(), "some_valid_openshift_resource", tok.Raw)
+		// then
+		require.NoError(t, err)
+		assert.Equal(t, "user_foo", username)
+		assert.Equal(t, "an_openshift_token", accessToken)
+	})
+
+	t.Run("invalid resource", func(t *testing.T) {
+		// when
+		_, _, err := authService.ResolveUserToken(context.Background(), "some_invalid_resource", tok.Raw)
+		// then
+		require.Error(t, err)
```

Comment:
Check error message too to make sure you got the expected error. Here and in other places too.
Maybe we can benefit from a function like this https://github.com/fabric8-services/fabric8-auth/blob/master/test/error.go#L11
Usage: https://github.com/fabric8-services/fabric8-auth/blob/4a020fbf23eff0b479c6a98d2a8f8452ab6f2c57/authorization/space/service/space_service_blackbox_test.go#L28